### PR TITLE
Apply master (v1) latest version to v2

### PR DESCRIPTION
**Pull request by gabrielcorado**
The following is being ported: #139, #143, #148, #151, #150 and #154. 

The latest addition to master (#153) is not being ported in this PR. It will require some changes and will be done in a separate PR.

### CODE REVIEW

File: service/remote_test.go [hunk 1]
```diff
@@ -68,6 +69,12 @@ func (m *MyComp) RemoteErr(ctx context.Context) (*test.SomeStruct, error) {
 
 type unregisteredStruct struct{}
 
+func errorIs(t *testing.T, err1 error, err2 error)  {
+	if !assert.ObjectsAreEqual(err1, err2) && !errors.Is(err1, err2) {
```
Comment (DylanGuedes) [inline service/remote_test.go:73]: Shouldn't it be `!assert.ObjectsAreEqual(err1, err2) || !errors.Is(err1, err2)`?

**Comment by victor-carvalho** [comment]
LGTM

**Review by victor-carvalho — APPROVED** [review]

**Review by rodopoulos — APPROVED** [review]
LGTM

File: service/remote_test.go (hunk 1 above)
Comment (gabrielcorado) [reply service/remote_test.go:73]: I think we assume that the struct being equal is not enough to consider it as the same (same for the opposite). But @victor-carvalho has more context on it.

Comment (DylanGuedes) [reply service/remote_test.go:73]: Gotcha. Just double checking, anyway 😂

**Pull request merged**